openpype/hosts/tvpaint/plugins/publish/collect_instance_frames.py

```diff
@@ -34,3 +34,111 @@ def process(self, instance):
                 frame_start, frame_end, instance.data["subset"]
             )
         )
+
+        custom_instance_frames = instance.data["creator_attributes"].get("custom_frames", None)
+        keep_frame_index = instance.data["creator_attributes"].get("keep_frame_index", False)
+
+        # If we want to keep the frame index from the tvpp scene and not recalculate them
+        if keep_frame_index:
+            frame_start = (instance.context.data["sceneMarkIn"] + instance.context.data["sceneStartFrame"])
+            frame_end = (instance.context.data["sceneMarkOut"] + instance.context.data["sceneStartFrame"])
+            instance.data["keepFrameIndex"] = keep_frame_index
+            instance.data["frameStart"] = frame_start
+            instance.data["frameEnd"] = frame_end
+
+        # If custom frames are given
+        if custom_instance_frames:
+            # Create a list of custom frame to render
+            custom_frames = self.range_custom_frames(
+                        custom_instance_frames,
+                        context.data["sceneMarkIn"],
+                        context.data["sceneMarkOut"],
+                        context.data["sceneStartFrame"]
+                        )
+            # Avoid exporting frame before the tracker frameStart
+            if not all(frame >= asset_doc["data"]["frameStart"] for frame in custom_frames):
+                self.log.warning("The custom frames to export start BEFORE the scene Tracking Start Frame")
+                self.log.info("An auto clean will be applied to start at {}".format(asset_doc["data"]["frameStart"]))
+                # Remove frames lower that the tracker frameStart
+                custom_frames = [frame for frame in custom_frames if (asset_doc["data"]["frameStart"] <= frame)]
+                # Replace by the true tracker frameStart
+                custom_frames.insert(0, asset_doc["data"]["frameStart"])
+
+            # Avoid exporting frame before the scene sceneStartFrame
+            if not all(frame >= instance.context.data["sceneStartFrame"] for frame in custom_frames):
```

Comment:
I think you should simply check the greatest (biggest) frame start value between `asset_doc["data"]["frameStart"]` and `instance.context.data["sceneStartFrame"]` and use this value to trim the frame lower than this frame value, this avoid doing twice trhe same operation

Reply:
ah oui en effet, je vais refacto cela